src/MusicStore/Components/ISystemClock.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+
+namespace MusicStore.Components
+{
+    /// <summary>
+    /// Abstracts the system clock to facilitate testing.
+    /// </summary>
+    public interface ISystemClock
```

Comment:
Not sure I understand what's wrong with the name... can you clarify that?

As far as switching between DateTime and DateTimeOffset, if we do such a change, we'd want to do it in the entire project. That might be better suited as a separate change and for now have this clock type use whatever date/time concept is already used in the app.